src/decision/dMCDA.jl

```diff
@@ -786,3 +786,31 @@ function unguided_site_selection(prefseedsites, prefshadesites, seed_years, shad
 
     return prefseedsites[prefseedsites.>0], prefshadesites[prefshadesites.>0]
 end
+
+
+"""
+    env_mean(env_layer::AbstractArray, dims_agg::Union{Symbol,Tuple{Symbol,Symbol},Int,Tuple{Int,Int}})
+
+Calculates mean over specified dimensions plus half the standard deviation.
+
+# Arguments
+- `env_layer` : Environmental data layer to calculate the mean of.
+- `dims_agg` : Dimensions to aggregate over.
+"""
+function env_mean(env_layer::AbstractArray, dims_agg::Union{Symbol,Tuple{Symbol,Symbol},Int,Tuple{Int,Int}})
+    return vec((mean(env_layer, dims=dims_agg) .+ std(env_layer, dims=dims_agg)) .* 0.5)
+end
```

Comment:
`env_mean()` isn't a very clear name for this - it does more than just take the mean.

@Zapiano any suggestions here?

Reply:
I'm also not comfortable with name and usage of `dims_agg`.

If `env_layer` has named dimensions, we should preference using those instead of numeric IDs for clarity.

Reply:
@Rosejoycrocker you are passing the result of this function to `site_selection` and the description of these parameters there are: "array of length nsites containing wave/dhw scenario.". It is not clear to me what do you mean by "length nsites". Maybe answering that could help us think of a better name for this function...?

Reply:
You could call it `mean_var_offset`..? And also, does it make sense for this parameter `0.5` to be passed as an argument with default value equals `0.5`? (maybe this doesn't make sense in this context, I thought about it because it's an arbitrary number)

Reply:
> I'm also not comfortable with name and usage of `dims_agg`, as:
> 
> * `dims` is the convention.
> * If `env_layer` has named dimensions, we should preference using those instead of numeric IDs for clarity.
> * If `env_layer` is always a named dimension, then constrain the type further to `NamedDimsArray`.

I'll see if env_layer in scenario.jl in actually a NamedDimsArray, I just copied the dimensions already used, but might be able to get rid of the input.

Reply:
> @Rosejoycrocker you are passing the result of this function to `site_selection` and the description of these parameters there are: "array of length nsites containing wave/dhw scenario.". It is not clear to me what do you mean by "length nsites". Maybe answering that could help us think of a better name for this function...?

Good point, they're basically a representation of dhws/waves that don't require looping over timesteps and performing site selection for each replicate of the dhws/waves. `mean_var_offset` could work... We only use 0.5 currently, but could be good to have the option to use other values.

Reply:
> `mean_var_offset` could work... We only use 0.5 currently, but could be good to have the option to use other values.

It's not an offset though. It summarizes the expected conditions $n$ number of years into the future.

Reply:
Oh and about the `0.5` it's just an equal weight between the mean and stdev, as I did this in a hurry.

If you want to consider it as something that can be adjusted, it would make more sense to flesh it out properly:

```julia
w = 0.5

vec((mean(env_layer, dims=dims_agg) .* w) .+ (std(env_layer, dims=dims_agg) .* (1.0 - w)))
```